drivers/led_strip/ws2812_i2s.c

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (c) 2022 Jonathan Rico
+ *
+ * Adapted from the SPI driver, using the procedure in this blog post:
+ * https://electronut.in/nrf52-i2s-ws2812/
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT worldsemi_ws2812_i2s
+
+#include <zephyr/drivers/led_strip.h>
+
+#include <string.h>
+
+#define LOG_LEVEL CONFIG_LED_STRIP_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ws2812_i2s);
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2s.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/dt-bindings/led/led.h>
+
+#define WS2812_I2S_PRE_DELAY_WORDS 1
+#define WS2812_I2S_LRCK_HZ 100000
+
+struct ws2812_i2s_cfg {
+    struct device const *dev;
+    size_t tx_buf_bytes;
+    struct k_mem_slab *mem_slab;
+    uint8_t num_colors;
+    const uint8_t *color_mapping;
+    uint16_t reset_words;
+    bool active_low;
+};
+
+/* Serialize an 8-bit clor channel value into two 16-bit I2S words (or 1 32-bit
+ * word).
+ */
+static inline void ws2812_i2s_ser(uint32_t *word, uint8_t color,
+                                  const uint8_t sym_one, const uint8_t sym_zero)
+{
+    *word = 0;
+    for (int i=0; i<8; i++) {
+        if ((1 << i) & color) {
+            *word |= sym_one << (i * 4);
+        } else {
+            *word |= sym_zero << (i * 4);
+        }
+    }
+
+    /* Swap the two I2S words due to the (audio) channel TX order. */
+    *word = (*word >> 16) | (*word << 16);
+}
+
+static int ws2812_strip_update_rgb(const struct device *dev,
+                                   struct led_rgb *pixels,
+                                   size_t num_pixels)
+{
+    const struct ws2812_i2s_cfg *cfg = dev->config;
+    uint8_t sym_one, sym_zero;
+    uint32_t reset_word;
+    uint32_t *tx_buf;
+    uint32_t flush_time_us;
+    void *mem_block;
+    int ret;
+
+    if(cfg->active_low) {
+        sym_one = 0x1;
+        sym_zero = 0x7;
+        reset_word = 0xFFFFFFFF;
+    } else {
+        sym_one = 0xE;
+        sym_zero = 0x8;
+        reset_word = 0;
+    }
```

Comment:
ah so that's what it's for! I thought it was just to correct for the SPI timings that might differ a bit from manufacturer to manufacturer. 
Yeah I'll add `nibble-one` and `nibble-zero` properties (since it's 4-bits per symbol). I'll still keep the `active-low` prop though because that is useful for governing the reset and idle logic values.

Reply:
That's also correct. The SPI frequencies available can be different from a controller to another. So it serves both purposes.

Reply:
OK. With the current static configuration the timings are almost perfect for the Everlight B1414 controller:
```
0 bit: 300 ns high and 900 ns low.
1 bit: 900 ns high and 300 ns low.
```
While I still think it is interesting to allow the user to configure the frames of the 0 and 1 bits and the clock frequency (because there are probably almost compatible devices with other specifications out there), I also think it can be added later. So I won't block this PR if you don't add it now.